socket_containernet_testing.py

```diff
@@ -0,0 +1,86 @@
+import logging
+import sys
+import time
+from envs.utils import traffic_generator
+import docker
+
+logging.basicConfig(level=logging.INFO)
+
+def connect_containers():
+    docker_client = docker.from_env()
+    logging.info("Connecting to Containernet Background Traffic Gen")
+    try:
+        docker_traffic_gen = docker_client.containers.get("mn.lh2")
+    except docker.errors.NotFound as e:
+        logging.info("Docker background traffic generator not found, new attempt...")
+    except docker.errors.APIError as e:
+        logging.info("Server error background traffic generator, new attempt...")
+
+    time.sleep(1)
+
+    logging.info("Connecting to Containernet Background Traffic Receiver")
+    try:
+        docker_traffic_receiver = docker_client.containers.get("mn.rh2")
+    except docker.errors.NotFound as e:
+        logging.info("Docker traffic receiver not found, new attempt...")
+    except docker.errors.APIError as e:
+        logging.info("Server error traffic receiver, new attempt...")
+
+
+    logging.info("Connecting to Containernet Sender")
+    try:
+        docker_file_sender = docker_client.containers.get("mn.lh1")
+    except docker.errors.NotFound as e:
+        logging.info("Docker traffic receiver not found, new attempt...")
+    except docker.errors.APIError as e:
+        logging.info("Server error traffic receiver, new attempt...")
+
+    traffic_receiver_ip: str = "10.0.2.2",
+
+    return docker_traffic_gen, docker_traffic_receiver, docker_file_sender, traffic_receiver_ip
+    
+    
+def _start_traffic_generator(traffic_script, docker_traffic_gen):
+    logging.info("Starting Background traffic")
+    logging.info(f"Script used: {traffic_script}")
+    docker_traffic_gen.exec_run(f"./mgen {traffic_script}", detach=True)
+
+def _start_traffic_receiver(docker_traffic_receiver):
+    logging.info("Starting traffic receiver")
+    docker_traffic_receiver.exec_run('./mgen event "listen udp 4311,4312,4600" event "listen tcp 5311,5312"', detach=True)
+    logging.info(f"Receiver started!")
+
+def _start_background_traffic(traffic_script, docker_traffic_gen, docker_traffic_receiver):
+    _start_traffic_receiver(docker_traffic_gen)
```

Comment:
:warning:  You are setting the traffic generator as the traffic receiver, that's why your tests were going that quickly.